src/components/CodeSearchResults/CodeSearchResultsTable.jsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Icon } from '@edx/paragon';
+
+import TableContainer from '../../containers/TableContainer';
+import RemindButton from '../RemindButton';
+import RevokeButton from '../RevokeButton';
+
+import EcommerceApiService from '../../data/services/EcommerceApiService';
+
+const tableColumns = [
+  {
+    label: 'Coupon Batch',
+    key: 'couponName',
+  },
+  {
+    label: 'Code',
+    key: 'code',
+  },
+  {
+    label: 'Redeemed',
+    key: 'isRedeemed',
+  },
+  {
+    label: 'Redemption Date',
+    key: 'redemptionDate',
+  },
+  {
+    label: 'Course Title',
+    key: 'courseTitle',
+  },
+  {
+    label: 'Actions',
+    key: 'actions',
+  },
+];
+
+/**
+ * The search API response contains a record for each coupon batch a
```

Comment:
The API response returns records around "vouchers" that exist, and it is possible (and likely) that a particular coupon batch can have multiple vouchers. That said, we'll know if vouchers are part of the same coupon batch based on the `coupon_id` returned in the response. Suggestion for changing the language:

`The search API response contains records for all vouchers a user/user_email is associated with (i.e., has unused assignments, or actual redemptions). To get a complete grouping of assignments or redemptions for a particular coupon batch for the user, we need to combine the results in assignments_and_redemptions (which contains all the metadata associated with a redemption, if applicable) for all vouchers that share the same coupon_id field.` Because we want the search results UI to show a row for each unused assignment and all redeemed courses associated with a particular user, this function flattens the search results by created a new table row for each value in each redemptions_and_assignments field.`